File: poo/python/Heranca/ContaCorrente.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 06/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Conta import Conta

class ContaCorrente(Conta):

    # Construtor #
    def __init__(self, num=0):
        super(ContaCorrente, self).__init__(num)
    
    # Redefinida #
    def atualiza(self, taxa):
        self._saldo += 2*(self._saldo * taxa)

    # Redefinida #
    def credita(self, quantia):
        if quantia > 0:
            self._saldo += quantia
            self._saldo -= 0.1
        else:
            raise ValueError('Não se pode depositar uma quantia negativa')

File: poo/java/Exceptions/Exercicio1/ContaCorrente.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class ContaCorrente extends Conta{

    /* Construtor */
    public ContaCorrente(int numero){
        this.numero = numero;
        this.saldo = 0;
    }

    public void atualiza(double taxa){
        this.saldo += 2*(this.saldo * taxa);
    }

    @Override
    public void credita(double quantia) throws IllegalArgumentException{
        if (quantia > 0){
            this.saldo += quantia;
            this.saldo -= 0.10;
        }
        else{
            throw new IllegalArgumentException("Não se pode depositar uma quantia negativa");
        }
    }
}
